src/core/net/udp6.cpp

```diff
@@ -297,12 +342,40 @@ otError Udp::SendTo(SocketHandle &aSocket, Message &aMessage, const MessageInfo
 
 void Udp::AddSocket(SocketHandle &aSocket)
 {
-    IgnoreError(mSockets.Add(aSocket));
+    SuccessOrExit(mSockets.Add(aSocket));
+
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+    if (mPrevBackboneSockets == nullptr)
+    {
+        mPrevBackboneSockets = &aSocket;
+    }
+#endif
+exit:
+    return;
 }
 
 void Udp::RemoveSocket(SocketHandle &aSocket)
 {
     SuccessOrExit(mSockets.Remove(aSocket));
```

Comment:
I think we can do a small optimization here.
 
If instead of `Remove()` we first search for entry in list (using `Find()`) we get the `prev` entry before it, we can then remove it using `PopAfter()` and use it to see update `mPrevBackboneSockets` (if needed).
```
     SocketHandle *prev;

     SuccessOrExit(mSockets.Find(aSocket , prev));
     mSocket.PopAfter(prev);
     
 #if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
     if (&aSocket == mPrevBackboneSockets)
     {
           mPrevBackboneSockets = prev;
     }
#endif 
```

Reply:
I didn't put much thought into it. The code you suggested is indeed more efficient. Thanks.